doc/source/building/build_in_buildservice.rst

```diff
@@ -1,27 +1,167 @@
+
+
+Advantages of using the Open Build Service (OBS)
+------------------------------------------------
+
+The Open Build Service offers multiple advantages over running KIWI
+locally:
+
+* OBS will host the latest successful build for you without having to setup
+  a server yourself.
+
+* As KIWI is fully integrated into OBS, OBS will automatically rebuild your
+  images if one of the included packages or one of its dependencies or KIWI
+  itself get updated.
+
+* The builds will no longer have to be executed on your own machine, but
+  will run on OBS, thereby saving you resources. Nevertheless, if a build
+  fails, you get a notification via email (if enabled in your user's
+  preferences).
+
+
+Differences Between Building Locally and on OBS
+-----------------------------------------------
+
+Note, there is a number of differences when building images with KIWI using
+the Open Build Service. Your image that build locally just fine, might not
+build without modifications.
+
+The notable differences to running KIWI locally include:
+
+* OBS will pick the KIWI package from the available repositories that you
+  specified in :file:`config.xml`/:file:`config.kiwi`. This means, that you
+  will be very likely building with a different KIWI version in the Open
+  Build Service.
+  This is especially relevant when building images for older versions like
+  SUSE Linux Enterprise. Therefore, include the custom appliances
+  repository as described in the following section:
+  :ref:`obs-recommended-settings`.
+
+* When KIWI runs on OBS, OBS will extract the list of packages from
+  :file:`config.xml` and use it to create a build root. In contrast to a
+  local build (where your distributions package manager will resolve the
+  dependencies and install the packages), OBS will **not** build your image
+  if there are multiple packages that could be chosen to satisfy the
+  dependencies of your packages. This manifests as errors like this:
+
+  .. code:: bash
+
+     unresolvable: have choice for SOMEPACKAGE: SOMEPAKAGE_1 SOMEPACKAGE_2
+
+  This can be solved by explicitly specifying one of the two packages in
+  the project configuration via the following setting:
+
+  .. code:: bash
+
+     Prefer: SOMEPACKAGE_1
+
+  Place the above line into the project configuration, which can be
+  accessed either via the web interface (click on the tab ``Project
+  Config`` on your project's main page) or via ``osc meta -e prjconf``.
+
+* OBS can only build a single image type, for example it cannot build a VMX
+  disk and a live ISO from one :file:`config.xml`. A configuration file
+  that contains multiple image types will therefore not work in the Open
+  Build Service.
+  Split the configuration file into two (or more) :file:`.kiwi` files and
+  provide them as separate packages inside your project.
+
+* OBS projects do not support subfolders, thus you must provide the overlay
+  files packed as an archive called :file:`root.tar.gz`.
```

Comment:
There's actually one more very important restriction:

* KIWI image builds in OBS cannot use `namedCollection` items, as the metadata backing those are not available. This is quite obvious when you try to use it to install groups for RH/Fedora image builds.

Reply:
I would add something like

```
* OBS projects do not support the use of the namedCollection section.
  A specification of ``` <namedCollection name="collection_name"/>``` is used to pass the
  information to install a collection of packages to the used package manager. The package
  manager can resolve that information only if the repository metadata contains the information
  about that collection and its packages. If the buildservice builds the image it resolves the given
  package list using its own sat solver based solver. From the result the buildservice creates
  temporary repositories of the same names as they got configured in the kiwi XML description.
  Those repositories doesn't contain the metadata to resolve collections.

  Even though kiwi uses the package manager to register the repositories, one should understand
  that the repositories the buildservice creates are just a subset of the data the real
  repositories provides.

  The dependency resolver mechanism the buildservice applies to create that repositories before
  kiwi is called is also different compared to the dependency resolution the selected package 
  manager performs later when kiwi calls it:

  * In the buildservice the order of repositories in the XML description matters
  * In the buildservice package dependencies from file provides are not resolved

  Because of that reason an image build that resolves well outside of the buildservice might
  not resolve when moving the image build into the buildservice
```

Reply:
Maybe this can be written nicer, or in sub paragraphs but I think that information is good to have here

Reply:
I have added the text in a slightly modified form: I have split up some of the sentences and formulated some of them differently.

Could you double check that it's still correct? I don't really understand most of that (but the behavior sound pretty, let's call it, odd tbh...).